sympy/physics/control/lti.py

```diff
@@ -1422,3 +1424,326 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+    [                             4          ]
+    [   -3          -a - s     - p  + 3*p - 2]
+    [  -----      ----------   --------------]
+    [  s + 2       2               p + s     ]
+    [             s  + s + 1                 ]
+
+    >>> tf_5 = TransferFunction(5, s, s)
+    >>> tf_6 = TransferFunction(5*s, (2 + s**2), s)
+    >>> tf_7 = TransferFunction(5, (s*(2 + s**2)), s)
+    >>> tf_8 = TransferFunction(5, 1, s)
+    >>> tfm_3 = TransferFunctionMatrix([[tf_5, tf_6], [tf_7, tf_8]])
+    >>> tfm_3
+    TransferFunctionMatrix(((TransferFunction(5, s, s), TransferFunction(5*s, s**2 + 2, s)), (TransferFunction(5, s*(s**2 + 2), s), TransferFunction(5, 1, s))))
+    >>> pprint(tfm_3, use_unicode=False)
+    [    5        5*s  ]
+    [    -       ------]
+    [    s        2    ]
+    [            s  + 2]
+    [                  ]
+    [    5         5   ]
+    [----------    -   ]
+    [  / 2    \    1   ]
+    [s*\s  + 2/        ]
+    >>> tfm_3.var
+    s
+    >>> tfm_3.shape
+    (2, 2)
+    >>> tfm_3.num_outputs
+    2
+    >>> tfm_3.num_inputs
+    2
+    >>> tfm_3.args
+    (((TransferFunction(5, s, s), TransferFunction(5*s, s**2 + 2, s)), (TransferFunction(5, s*(s**2 + 2), s), TransferFunction(5, 1, s))),)
+
+    To access the ``TransferFunction`` at any index in the ``TransferFunctionMatrix``, use the index notation.
+
+    >>> tfm_3[1, 0]  # gives the TransferFunction present at 2nd Row and 1st Col. Similar to that in Matrix classes
+    TransferFunction(5, s*(s**2 + 2), s)
+    >>> tfm_3[0, 0]  # gives the TransferFunction present at 1st Row and 1st Col.
+    TransferFunction(5, s, s)
+    >>> tfm_3[:, 0]  # gives the first column
+    TransferFunctionMatrix(((TransferFunction(5, s, s),), (TransferFunction(5, s*(s**2 + 2), s),)))
+    >>> pprint(_, use_unicode=False)
+    [    5     ]
+    [    -     ]
+    [    s     ]
+    [          ]
+    [    5     ]
+    [----------]
+    [  / 2    \]
+    [s*\s  + 2/]
+    >>> tfm_3[0, :]  # gives the first row
+    TransferFunctionMatrix(((TransferFunction(5, s, s), TransferFunction(5*s, s**2 + 2, s)),))
+    >>> pprint(_, use_unicode=False)
+    [5   5*s  ]
+    [-  ------]
+    [s   2    ]
+    [   s  + 2]
+
+    To negate a transfer function matrix, ``-`` operator can be prepended:
+
+    >>> tfm_4 = TransferFunctionMatrix([[tf_2], [-tf_1], [tf_3]])
+    >>> -tfm_4
+    TransferFunctionMatrix(((TransferFunction(-p**4 + 3*p - 2, p + s, s),), (TransferFunction(a + s, s**2 + s + 1, s),), (TransferFunction(-3, s + 2, s),)))
+    >>> tfm_5 = TransferFunctionMatrix([[tf_1, tf_2], [tf_3, -tf_1]])
+    >>> -tfm_5
+    TransferFunctionMatrix(((TransferFunction(-a - s, s**2 + s + 1, s), TransferFunction(-p**4 + 3*p - 2, p + s, s)), (TransferFunction(-3, s + 2, s), TransferFunction(a + s, s**2 + s + 1, s))))
+
+    ``subs()`` returns an ImmutableDenseMatrix object with the value substituted in the expression. This will not
+    mutate your original ``TransferFunctionMatrix``.
```

Comment:
Based on the constructor down below, I suppose `TransferFunctionMatrix.subs(some_variable, value)` should just work. Var substitution in a transfer function returns a transfer function; TFM must follow the same principle, no?

We don't yet allow the substitution of `var` in `TransferFunction`;
```python
In [19]: G = TransferFunction(x - s, x**2 + 1, s)                      

In [20]: G.subs(x, 2)                                                  
Out[20]: TransferFunction(2 - s, 5, s)

In [21]: G.subs({s: 2, x: 2})                                          
Out[21]: TransferFunction(2 - s, 5, s)
```
But your example here (for TFM) shows otherwise.

Reply:
I will remove the existing `subs` method then and update the example.

Reply:
> We don't yet allow the substitution of `var` in `TransferFunction`;

Although, I think we might need this while evaluating a transfer function over a range of frequencies.

Reply:
> We don't yet allow the substitution of `var` in `TransferFunction`;

Why doesn't that work?

Reply:
See https://github.com/sympy/sympy/pull/19390#issuecomment-640556483.